pkg/splunk/enterprise/searchheadcluster.go

```diff
@@ -173,17 +173,18 @@ func ApplySearchHeadCluster(client splcommon.ControllerClient, cr *enterpriseApi
 	}
 	cr.Status.Phase = phase
 
-	if cr.Status.AppContext.AppsSrcDeployStatus != nil && cr.Status.DeployerPhase == splcommon.PhaseReady {
-		markAppsStatusToComplete(client, cr, &cr.Spec.AppFrameworkConfig, cr.Status.AppContext.AppsSrcDeployStatus)
-		// Schedule one more reconcile in next 5 seconds, just to cover any latest app framework config changes
-		if cr.Status.AppContext.IsDeploymentInProgress {
-			cr.Status.AppContext.IsDeploymentInProgress = false
-			return result, nil
-		}
-	}
-
 	// no need to requeue if everything is ready
 	if cr.Status.Phase == splcommon.PhaseReady {
+		if cr.Status.AppContext.AppsSrcDeployStatus != nil && cr.Status.DeployerPhase == splcommon.PhaseReady {
```

Comment:
This is put here because we should only push the bundle once both Deployer and SHC members are ready or else we would keep trying and fail to push the bundle until the SHC members come up.

Reply:
Can we move this check to the playbook, such that the local installs are not blocked? Probably, just exiting the playbook.